compiler/noirc_evaluator/src/ssa/opt/inline_simple_functions.rs

```diff
@@ -1,6 +1,11 @@
-//! This modules defines an SSA pass that inlines calls to functions with at most one instruction.
+//! This modules defines an SSA pass that inlines calls to simple functions.
 //! That is, the contents of the called function is put directly into the caller's body.
 //! Functions are still restricted to not be inlined if they are recursive or marked with no predicates.
+//!
+//! A simple function is defined as the following:
+//! - Contains no more than 10 instructions
+//! - The function only has a single block (e.g. no control flow or conditional branches)
+//! - It is not marked with the no predicates [inline type][noirc_frontend::monomorphization::ast::InlineType]
```

Comment:
It would be great to include an explainer about why "no predicates" are treated separately, or a link to any existing documentation about it. There is some description on `InlineType::NoPredicate`, but it only states that it's inlined later, without explaining why.

Reply:
Awesome, thanks!